ppb_vector/vector2.py

```diff
@@ -114,4 +114,6 @@ def reflect(self, surface_normal: 'Vector2') -> 'Vector2':
         """
         if not (0.99999 < surface_normal.length < 1.00001):
             raise ValueError("Reflection requires a normalized vector.")
+        if self * surface_normal > 0:
```

Comment:
We're calculating the dot-product twice here. Here and on line 119.

Would love if we can only calculate once.

Reply:
I was wrong about this. Please ignore.